File: app/src/main/java/com/doubean/ford/data/vo/PostComment.java
package com.doubean.ford.data.vo;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

import com.doubean.ford.data.db.Converters;
import com.google.gson.annotations.SerializedName;

import java.time.LocalDateTime;
import java.util.List;

@Entity(tableName = "post_comments")
@TypeConverters(Converters.class)
public class PostComment extends Item {
    @PrimaryKey
    @NonNull
    public String id;

    public User author;

    @ColumnInfo(name = "post_id")
    public String postId;

    public List<SizedPhoto> photos;

    public String text;

    @SerializedName("create_time")
    public LocalDateTime created;

    @SerializedName("vote_count")
    public int voteCount;

    @SerializedName("ref_comment")
    public PostComment repliedTo;

    @Override
    public String getId() {
        return id;
    }

}

File: app/src/main/java/com/doubean/ford/data/vo/PostComment.kt
package com.doubean.ford.data.vo

import androidx.room.*
import com.doubean.ford.data.db.Converters
import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

@Entity(tableName = "post_comments")
@TypeConverters(Converters::class)
data class PostComment  (
    @PrimaryKey
    override val id: String,

    val author: User,

    val photos: List<SizedPhoto>? = null,

    val text: String? = null,

    @SerializedName("create_time")
    val created: LocalDateTime? = null,

    @SerializedName("vote_count")
    val voteCount: Int = 0,

    @SerializedName("ref_comment")
    val repliedTo: PostComment? = null

):Item(){
    @ColumnInfo(name = "post_id")
    lateinit var postId: String
}